plugins/module_utils/resource.py

```diff
@@ -50,14 +50,15 @@ def run(desired_state, current_state, client, state):
 
     sorter.prepare()
     with concurrent.futures.ThreadPoolExecutor() as executor:
+        current_state["changed"] = False
         while sorter:
             futures = {}
             for name in sorter.get_ready():
                 if state == "present":
                     node = resolve_refs(desired_state[name], current_state)
                     futures[executor.submit(client.present, node)] = name
                 elif state == "absent":
-                    if name not in current_state:
+                    if name in current_state:
```

Comment:
This change allows the resources to get deleted when state: absent, otherwise, .delete() is not called.

Reply:
But this change means the delete never happens. It just marks the resource as done in the sorter and moves on to the next one.

Reply:
Hmm...I tried tried this playbook with that change. I first created and then deleted switching state. I checked the AWS console before and after and it seems deleting it. Not sure then...

```
- hosts: localhost
  gather_facts: false
  
  tasks:
    - set_fact:
        resources:
          role_1:
            Type: AWS::IAM::Role
            Properties:
              RoleName: eda-test-role
              Description: event-driven test role
              AssumeRolePolicyDocument:
                Version: 2012-10-17
                Statement:
                  - Action:
                      - sts:AssumeRole
                    Effect: Allow
                    Principal:
                      Service:
                          - ec2.amazonaws.com

    - name: res_1
      cloud.pravic.resources:
        state: "absent"
      register: _result
          
```

Reply:
This is likely because you have no state file, and as a result `current_state` is always empty. The state file is only generated if the `cloud.pravic.state` callback plugin is enabled.